ledger-service/http-json/src/main/scala/com/digitalasset/http/WebSocketService.scala

```diff
@@ -300,6 +312,25 @@ class WebSocketService(
     }
   }
 
+  private def removePhantomArchives[A, B](remove: Boolean) =
+    if (remove) removePhantomArchives_[A, B]
+    else Flow[StepAndErrors[A, B]].map(identity)
+
+  private def removePhantomArchives_[A, B]
+    : Flow[StepAndErrors[A, B], StepAndErrors[A, B], NotUsed] =
+    Flow[StepAndErrors[A, B]]
+      .scan((Set.empty[String], Option.empty[StepAndErrors[A, B]])) {
+        case ((s0, _), a0) =>
+          val deletesToKeep: Set[String] = a0.step.deletes.filter(x => s0(x))
```

Comment:
```suggestion
          val deletesToKeep: Set[String] = a0.step.deletes intersect s0
```

Observing that the above is equal, and `intersect` is symmetric, you could also use partition

```suggestion
          val (deletesToKeep, s0md) = s0 partition a0.step.deletes
```

As `s0md` is `s0 -- deletesToKeep`. (As you may also have gathered, `partition` is the built-in "blind boolean" variant of our `partitionMap` utility.)